cmake/CPM.cmake

```diff
@@ -1071,7 +1095,24 @@ function(cpm_fetch_package PACKAGE populated)
   string(TOLOWER "${PACKAGE}" lower_case_name)
 
   if(NOT ${lower_case_name}_POPULATED)
-    FetchContent_Populate(${PACKAGE})
+    if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.28.0")
+      if(DOWNLOAD_ONLY)
+        # MakeAvailable will call add_subdirectory internally which is not what we want when
+        # DOWNLOAD_ONLY is set. Populate will only download the dependency without adding it to the
+        # build
+        FetchContent_Populate(
+          ${PACKAGE}
+          SOURCE_DIR "${CPM_FETCHCONTENT_BASE_DIR}/${lower_case_name}-src"
+          BINARY_DIR "${CPM_FETCHCONTENT_BASE_DIR}/${lower_case_name}-build"
+          SUBBUILD_DIR "${CPM_FETCHCONTENT_BASE_DIR}/${lower_case_name}-subbuild"
+          ${ARGN}
+        )
+      else()
+        FetchContent_MakeAvailable(${PACKAGE})
+      endif()
+    else()
+      FetchContent_Populate(${PACKAGE})
+    endif()
```

Comment:
Question out of curiosity as I now realise this form of `FetchContent_Populate` isn't deprecated: could we just change the original `FetchContent_Populate` here to the explicit syntax here and have it work on both old and new version of CMake without the version-specific changes? Or is there something I'm missing?

Reply:
Yes, we could switch fully over to `*_Populate`. 

I’m not really a fan of that, since it seems like the CMake devs are pushing us towards using `*_Declare` and `*_MakeAvailable` over the last few updates. Switching to `*_Populate` feels like we’re going against the trend.

That said, maintaining three different code paths to achieve the same goal is probably alot worse. It complicates the review process and makes the code harder to manage.

I guess, if we run into issues with `*_Populate` later, we can deal with them then. :D

Anyway, latest commit switches everything to `*_Populate` with the explicit syntax.  This has the sideeffect that `cpm_declare_fetch` could be removed, because it's not being used by anything.

The unittests passed locally, but I'm out of time testing on some of my larger projects. :D